text/0000-history.md

```diff
@@ -0,0 +1,85 @@
+# Meta
+[meta]: #meta
+- Name: Buildpack layer metadata
+- Start Date: 2021-12-02
+- Author(s): [@samj1912](https://github.com/samj1912)
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Add layer history metadata to the output image for buildpack and app layers to improve visualization in tools like `DockerHub`, `dive` etc.
+
+# Definitions
+[definitions]: #definitions
+
+`history`: The history key in the `OCI` config. See [properties](https://github.com/opencontainers/image-spec/blob/main/config.md#properties)
+
+# Motivation
+[motivation]: #motivation
+
+This will allow Buildpack built images to be better visualized by common container visualization and debugging tools.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Various container image introspection tools look at `history.created_by` for each layer to visualize the layer. We should populate this key with a value that describes where a layer came from, for buildpacks this can be -
+
+- `Layer: {{ buildpack.layer.name }}, Created by: {{ buildpack.id }}@{{ buildpack.version }}`
+
+for app layers, this can be - 
+
+- `Application Slice: {{ slice_number }} Created by: {{ buildpack.id }}@{{ buildpack.version }}`
```

Comment:
I'm assuming adding the `buildpack.version` here doesn't impact the concept of reproducible builds even when there's a minor buildpack version bump if the results of the buildpack execution yields the same outcome right?

Reply:
We already store this information as a label in the output image.